### ISSUE TITLE
Build in Ubuntu

### ISSUE THREAD

**Issue by AwePhD** [plp13/qman#3]
Thanks you for this program, it's more comfortable to read the man pages now :)

I would like to suggest to enhance the build instructions for Ubuntu, since there is a small tweak.

We need to install the packages `python-cogapp`, `libinih-dev` and `libinih1` (plus other missing dependencies, the arch `PKGBUILD` helped me a lot there).
Then `cog` is named `cogapp`. I do not know meson build tool, so I just created a symlink `sudo ln -s /usr/bin/cogapp /usr/bin/cog`. Maybe adding a target in the `build/ninja.build` is cleaner? Again, I am not aware how to use C build tools expect Makefile. Then the build can be done with `meson setup build/ src/` and `meson compile`. I did not test install. Ubuntu Noble (24.04)

**Comment by plp13** [comment]
Hello AwePhD,

Can you try going into `src/meson.build` and changing `cog = find_program('cog')` to `cog = find_program('cogapp')`?

Then proceed as normal:

```
$ meson setup build/ src/
$ cd build/
$ meson compile
```
Let me know if this works. I'd love to see Qman running on Debian or Ubuntu, however I have only have access to an Ubuntu 20 LTS box right now, which doesn't have the `python-cogapp` package at all.

Meson is really much simpler than Makefile. You just define your sources, targets, and dependencies by calling a handful of Python functions, and it takes care of everything else. Have a look at its [tutorial](https://mesonbuild.com/Tutorial.html).

**Comment by AwePhD** [comment]
The change in `meson.build` file works like a charm. No need of symbolic link, sweet. I do think a `python3 -m pip install cogapp` should do the work. Maybe 

Yep, python3-cogapp is a Python system package. 

The tutorial is very nice. Meson seems to be a great tool.

Plus, I made the build work on a focal [docker image](https://hub.docker.com/layers/library/ubuntu/focal-20240530/images/sha256-d86db849e59626d94f768c679aba441163c996caf7a3426f44924d0239ffe03f?context=explore). Notably I installed the following package: `build-essential libinih-dev libinih1 ninja-build meson python3-pip ncurses-base ncurses-bin ncurses-term libncursesw6 libncurses6 python3-pkgconfig` and run `pip install cogapp`, it adds the `cog` binary in my PATH, not cogapp. The build step is the same expect that we should use ninja directly, because the meson is verison 0.53.2 so there is build tool agnostic `compile` subcommand.
I am not sure that the `ncurses*` packages are minimal ...

**Comment by plp13** [comment]
OK. I set up an Ubuntu 24.04 container (the Canonical image) for testing.

Using that and your input, I managed to figure out a minimal list of packages that need to be installed, in order for the program to compile and run correctly. I added it to README.md for future reference.

I also modified `meson.build` to search for both `cog` and `cogapp` upon `meson setup`.

So, I think now the issue is solved, unless you disagree.
